src/lib/xmltok.rs

```diff
@@ -587,20 +587,20 @@ impl<T: NormalEncodingTable> XmlEncodingImpl for Utf8EncodingImpl<T> {
         from: &mut ExpatBufRef,
         to: &mut ExpatBufRefMut<u16>,
     ) -> XML_Convert_Result {
-        let mut current_block: u64;
         let mut res: XML_Convert_Result = XML_Convert_Result::COMPLETED;
         // let mut to: *mut c_ushort = *toP;
         // let mut from: *const c_char = *fromP;
         loop {
```

Comment:
Could rewrite this loop as `while !from.is_empty() && !to.is_empty()` (which is basically what the original C code does), and move the `from.is_empty` check after it. See https://github.com/libexpat/libexpat/blob/master/expat/lib/xmltok.c#L404

Reply:
Good point! Implemented in 5a09485.